airbyte-integrations/connectors/source-file/file_source/source.py

```diff
@@ -0,0 +1,241 @@
+"""
+MIT License
+
+Copyright (c) 2020 Airbyte
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+
+from datetime import datetime
+from typing import Generator, List
+
+import gcsfs
+import pandas as pd
+from airbyte_protocol import (
+    AirbyteCatalog,
+    AirbyteConnectionStatus,
+    AirbyteMessage,
+    AirbyteRecordMessage,
+    AirbyteStream,
+    Source,
+    Status,
+    Type,
+)
+from google.cloud.storage import Client
+from smart_open import smart_open
+
+
+class FileSource(Source):
+    """This source aims to provide support for readers of different file formats stored in various locations.
+
+    It is optionally using s3fs, gcfs or smart_open libraries to handle efficient streaming of very large files
+    (either compressed or not).
+
+    Supported examples of URL this can accept are as follows:
+    ```
+        s3://my_bucket/my_key
+        s3://my_key:my_secret@my_bucket/my_key
+        s3://my_key:my_secret@my_server:my_port@my_bucket/my_key
+        gs://my_bucket/my_blob
+        azure://my_bucket/my_blob
+        hdfs:///path/file
+        hdfs://path/file
+        webhdfs://host:port/path/file
+        ./local/path/file
```

Comment:
curious if the local paths actually work. see some description around the complexity of the local fs stuff [https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/destination-csv/src/main/resources/spec.json#L11](url). basically we only mount one directory so we need to provide the user with some help to understand how they can pull info from a local filesystem.

Reply:
it works in a Python script but you are right, I haven't tried it through the docker images yet...

Would it be unacceptable to add extra mount arguments to the docker run command?
if you can mount -v the "destination_path:destination_path" so the path is accessible inside the docker container as it is on the host

Reply:
i think that doesn't work well on back because there are limitations on what dirs can be mounted. that's why we set one local mount. I would suggest going this route: https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/destination-csv/src/main/resources/spec.json#L11. Where we just tell the user if you're using your local filesystem, it better start with /local and use the local mount. if you can find something better though, i'm down.

Reply:
I can look at this as part of a new issue?